### PR TITLE
Fix client error in regression

### PR DESCRIPTION
**Pull request by brendan-priorlabs**
For some reason we are (and apparently were?) getting `None` in the logits from regression with full output. Not clear to me why this would happen, but since it's a standing issue, seem worth just loosening the type

### CODE REVIEW

**Comment by brendan-priorlabs** [comment]
@oscarkey, could you sanity check that point about None logits? Seems weird!

**Comment by oscarkey** [comment]
this is the output of `TabPFNRegressor.predict()`?
This is a bit strange, I can't work out how `logits` could be None. We seem to call `logits.log()` just before adding it to the dict, so I assume this would crash if it was None: https://github.com/PriorLabs/TabPFN/blob/6bcd9b7a0b5bd55b653c0577048f95d79585bc19/src/tabpfn/regressor.py#L1013
and it seems like it's been like this for a while.

**Review by simo-prior — APPROVED** [review]

**Comment by simo-prior** [comment]
> this is the output of `TabPFNRegressor.predict()`? This is a bit strange, I can't work out how `logits` could be None. We seem to call `logits.log()` just before adding it to the dict, so I assume this would crash if it was None: https://github.com/PriorLabs/TabPFN/blob/6bcd9b7a0b5bd55b653c0577048f95d79585bc19/src/tabpfn/regressor.py#L1013 and it seems like it's been like this for a while.

@oscarkey we could have `inf`, `-inf`, and `nan`, those are all converted to `None` once serialized to json.

**Comment by oscarkey** [comment]
ahhh well that explains it then!

File: src/tabpfn_client/api_models.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
     List[PredictionScalar],
     List[float],
     List[List[float]],
-    Dict[str, Union[List[float], List[List[float]]]],
+    Dict[str, Union[List[Optional[float]], List[List[Optional[float]]]]],
```
[the remark this answers is not in the document]
Comment (brendan-priorlabs) [reply src/tabpfn_client/api_models.py:13]: quick_test.py works, so I think this is a red herring.

**Comment by brendan-priorlabs** [comment]
Thank you both!
